packages/server/src/workers/subscription.ts

```diff
@@ -126,6 +130,53 @@ export function getSubscriptionQueue(): Queue<SubscriptionJobData> | undefined {
   return queue;
 }
 
+/**
+ * Checks if this resource should create a notification for this `Subscription` based on the access policy that should be applied for this `Subscription`.
+ * The `AccessPolicy` of author's `ProjectMembership` for this resource's `Project` is used when evaluating whether the `AccessPolicy` is satisfied.
+ *
+ * Currently we log if the `AccessPolicy` is not satisfied only.
+ *
+ * TODO: Actually prevent notifications for `Subscriptions` where the `AccessPolicy` is not satisfied.
+ *
+ * @param resource - The resource to evaluate against the `AccessPolicy`.
+ * @param project - The project containing the resource.
+ * @param subscription - The `Subscription` to get the `AccessPolicy` for.
+ */
+async function checkAccessPolicy(resource: Resource, project: Project, subscription: Subscription): Promise<void> {
+  try {
+    // Check access policy
+    const subAuthor = subscription.meta?.author;
+    if (subAuthor) {
+      const membership = await findProjectMembership(project.id as string, subAuthor);
+      if (membership) {
+        const accessPolicy = await buildAccessPolicy(membership);
+        const satisfied = !!satisfiedAccessPolicy(resource, AccessPolicyInteraction.READ, accessPolicy);
+        if (!satisfied) {
+          globalLogger.warn(
+            `[Subscription Access Policy]: Access Policy not satisfied on '${getReferenceString(resource)}' for '${getReferenceString(project)}'`,
```

Comment:
Preference to include `getReferenceString(subscription)` rather than `getReferenceString(project)` - subscription is more explicit.